cmake/CompilerFeatureTest.cmake

```diff
@@ -0,0 +1,25 @@
+# # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Functions that determine compiler capabilities
+
+include(CheckCXXSourceCompiles)
+
+# Determines if the selected C++ compiler has ranges support.
+# Sets 'result_var' to whether support is detected.
+function(beman_check_range_support result_var)
+    # Check if the C++ standard is at least C++20 or later.
+    if(CMAKE_CXX_STANDARD LESS 20)
+        set(${result_var} FALSE PARENT_SCOPE)
+        return()
+    endif()
+
+    check_cxx_source_compiles(
+        "
+        #include <ranges> // C++20 ranges; note that __cpp_lib_ranges is not defined for all compilers
```

Comment:
What compiler doesn't define it?

BTW, when searching for `__cpp_lib_ranges` using GitHub code search I found [this alternative way to check](https://github.com/SparseBLAS/spblas-reference/blob/70e417eaadd9b12c1ff969e01276a8b69e88f10f/CMakeLists.txt#L45).

Reply:
I have adopted this in my latest contribution.